### PR TITLE
#716 Remove save button from stocktake

### LINKED ISSUE #716: Remove save button on stocktake detail view
Epic for 'removing the save button' on stock take detail view!

This is still using 100% mock server. The backend is still a bit away.

### PR DESCRIPTION
**Pull request by joshxg**
Fixes #716 

- Just does the stocktake itself for now. Will do the lines next.

### CODE REVIEW

File: packages/common/src/types/schema.ts [hunk 1]
```diff
@@ -3006,6 +3006,16 @@ export const StocktakeDocument = gql`
...
     `;
+export const UpdateStocktakeDocument = gql`
+    mutation updateStocktake($input: UpdateStocktakeInput!) {
+  updateStocktake(input: $input) {
+    ... on StocktakeNode {
+      __typename
+      id
+    }
+  }
+}
```
Comment (joshxg) [inline packages/common/src/types/schema.ts:3017]: I'm still not sure of the API for this, so have left as is.

File: packages/inventory/src/Stocktake/DetailView/ContentArea.tsx [hunk 2]
```diff
@@ -0,0 +1,21 @@
+// import React, { FC } from 'react';
```
Comment (joshxg) [inline packages/inventory/src/Stocktake/DetailView/ContentArea.tsx:1]: Will fill this out correctly in the next PR with the lines.

File: packages/inventory/src/Stocktake/DetailView/DetailView.tsx [hunk 3]
```diff
@@ -70,7 +69,7 @@ export const DetailView: FC = () => {
   const { hideDialog, showDialog, Modal } = useDialog({
     onClose: () => setModalState({ item: null, mode: ModalMode.Create }),
   });
-  const { draft, save, onChangeSortBy, sortBy } = useDraftStocktake();
+  const { draft, onChangeSortBy, sortBy } = useDraftStocktake();
```
Comment (joshxg) [inline packages/inventory/src/Stocktake/DetailView/DetailView.tsx:72]: Once the lines have been completed, can finish off removing this.

File: packages/inventory/src/Stocktake/DetailView/Toolbar.tsx [hunk 4]
```diff
@@ -7,28 +7,31 @@ import {
...
-      .map(selectedId => draft.lines.find(({ id }) => selectedId === id))
-      .filter(Boolean) as StocktakeItem[],
-  }));
+  // const { selectedRows } = useTableStore(state => ({
+  //   selectedRows: Object.keys(state.rowState)
+  //     .filter(id => state.rowState[id]?.isSelected)
+  //     .map(selectedId => draft.lines.find(({ id }) => selectedId === id))
+  //     .filter(Boolean) as StocktakeItem[],
+  // }));
+  const selectedRows: unknown[] = [];
```
Comment (joshxg) [inline packages/inventory/src/Stocktake/DetailView/Toolbar.tsx:34]: lines again :)

File: packages/inventory/src/Stocktake/api/api.ts [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+//     };
+//   };
+
+// const createDeleteStocktakeLineInput = (
+//   line: StocktakeLine
+// ): DeleteStocktakeLineInput => {
+//   return {
+//     ...line,
+//   };
+// };
```
Comment (joshxg) [inline packages/inventory/src/Stocktake/api/api.ts:73]: Commented out as I'll use this in the next PR

File: packages/inventory/src/Stocktake/DetailView/columns.ts [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+export const useStocktakeColumns = ({
+  sortBy,
+  onChangeSortBy,
+}: UseStocktakeColumnOptions) =>
+  useColumns<StocktakeItem>(
+    [
+      'itemCode',
+      'itemName',
+      'batch',
+      'expiryDate',
```
Comment (mark-prins) [inline packages/inventory/src/Stocktake/DetailView/columns.ts:26]: I've had a wee play and added an expiry date formatter to `ColumnDefinitionSetBuilder`. what do you say to something like this?

```
  expiryDate: {
    key: 'expiryDate',
    label: 'label.expiry',
    width: 200,
    format: ColumnFormat.Date,
    formatter: dateString =>
      dateString ? formatExpiryDate(new Date(dateString as string)) || '' : '',
  },
```

**Review by mark-prins — APPROVED** [review]
Sweet! 🍬 
looks good - I like the 'many hook' pattern, nice to see it come across to stocktakes 😁

File: packages/inventory/src/Stocktake/DetailView/columns.ts (hunk 6 above)
Comment (joshxg) [reply packages/inventory/src/Stocktake/DetailView/columns.ts:26]: Yup great idea - been avoiding the formatters a little as they're not type safe :(

**Pull request merged**